sequoia/methods/d3rlpy_methods/base.py

```diff
@@ -0,0 +1,148 @@
+from typing import Type, ClassVar, List, Tuple, Dict, Optional, Union
+
+import gym
+from d3rlpy.algos import *
+import numpy as np
+from d3rlpy.dataset import MDPDataset
+from gym import Space
+from gym.wrappers.record_episode_statistics import RecordEpisodeStatistics
+from sequoia import Method, Environment, Observations, Actions, Rewards, TraditionalRLSetting
+from sequoia.settings.offline_rl.setting import OfflineRLSetting
+
+
+class OfflineRLWrapper(gym.Wrapper):
+    def __init__(self, env):
+        super().__init__(env)
+        self.observation_space = env.observation_space.x
+
+    def reset(self):
+        observation = super().reset()
+        return observation.x
+
+    def step(self, action):
+        observation, reward, done, info = super().step(action)
+        return observation.x, reward.y, done, info
+
+
+class BaseOfflineRLMethod(Method, target_setting=OfflineRLSetting):
+    Algo: ClassVar[Type[AlgoBase]] = AlgoBase
+
+    def __init__(self,
+                 train_steps: int = 1_000_000,
+                 train_steps_per_epoch=1_000_000,
+                 test_steps=1_000,
+                 scorers: dict = None,
+                 use_gpu: bool = False, **kwargs):
+        super().__init__()
+        self.train_steps = train_steps
+        self.train_steps_per_epoch = train_steps_per_epoch
+        self.test_steps = test_steps
+        self.scorers = scorers
+        self.offline_metrics = None
+
+        # Option 1: Move this to configure, each algo implements its own configure
+        # Option 2: Add kwargs to this constructor
+        #
+        # TODO: add support for all algo params, not just common ones
+
+        self.algo = type(self).Algo(use_gpu=use_gpu, **kwargs)
```

Comment:
Move this to `configure`, so that we don't reuse the same algorithm instance across different settings.

Reply:
You can save the kwargs on self or something like that if you want.

Reply:
done